Source/settings/Settings.js

```diff
@@ -9,6 +9,98 @@ function getPluginIdentifier() {
   return __command.pluginBundle().identifier()
 }
 
+// check if the argument is a native sketch object
```

Comment:
so I didn't properly documented yet so it's my bad but there are some packages shipping with sketch and we should take advantage of them instead of rewriting that:

```
const util = require('util')

function coerceNSObject(value) {
  if (value) {
    if (util. isObject(value)) {
      return util.toObject(value)
    } else if (util.isArray(value)) {
      return util.toArray(value)
    } else if (util.isString(value)) {
      return String(value)
    } else if (util.isNumber(value)) {
      return Number(value)
    }
  }
  return value
}
```

Reply:
I agree this could be nice to expose to plugins. I'm not sure which util package you are referring to. I'm unable to build with `require('util')`.

Also, forgive any obvious questions. I'm still fitting my JS hat back on after a long hiatus.

Reply:
indeed, you need to tell webpack that `util` is an external dependency: In the webpack.config.js file, you need to add
```js
const CORE_MODULES = Object.keys(
  require('./core-modules/package.json').dependencies
).map(k => k.replace('skpm/', ''))

...
const config = {
  externals: [
    (context, request, callback) => {
      // core modules shipped in Sketch
      if (CORE_MODULES.indexOf(request) !== -1) {
        return callback(null, `commonjs ${request}`)
      }
      return callback()
    },
  ],
...
```

Reply:
👍 Got that working, thanks. 

As for using that, there are a couple issues I'm fining with the functions in the util conversions. 

1. Object and Array don't do deep conversions
2. NSNumber with a bool is type NSCFBoolean which is not handled

Reply:
> Object and Array don't do deep conversions

Indeed, we'd still need that. I though the replacer in the JSON.stringify was called recursively? Maybe it's only for objects that aren't supplied by a previous replacer.

> NSNumber with a bool is type NSCFBoolean which is not handled

Good catch, I'll change the util package to handle that case.
The coerce method will need another case:
```js
 else if (util.isBoolean(value)) {
      return Boolean(Number(value))
    }
```